esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2084,45 +2084,46 @@ void WsWuInfo::readWorkunitLog(IFile* sourceFile, MemoryBuffer& buf, const char*
     else
         eof = lineReader->readLine(line.clear());
 
-    const unsigned positionProcessId = getPositionOfField(logfields, MSGFIELD_process);
+    const unsigned columnNumPID = getPositionOfField(logfields, MSGFIELD_process);
     bool outputThisLine = false;
     unsigned processID = 0;
     bool foundEndWUID = false; 
     while (!eof)
     {
-        if (line.length() > positionProcessId+1)
-        {
-            const char * lineStartingProcessId = line.str()+positionProcessId;
-            if (outputThisLine)
-            {
-                //If the slave is restarted before WU is finished, we cannot find out the "Finished wuid=...".
-                //So, we should check whether the slave is restarting or not.
-                unsigned pID = 0;
-                foundEndWUID = parseLogLine(lineStartingProcessId, endwuid, pID);
-                if ((pID > 0) && (pID != processID))
-                    break;
-            }
-            else if (strstr(lineStartingProcessId, startwuid))
-            {
-                outputThisLine = true;
-                foundEndWUID = false;
-                if (processID == 0)
-                    parseLogLine(lineStartingProcessId, nullptr, processID);
-            }
-        }
         if (outputThisLine)
         {
+            //If the slave is restarted before WU is finished, we cannot find out the "Finished wuid=...".
+            //So, we should check whether the slave is restarting or not.
+            unsigned pID = 0;
+            foundEndWUID = parseLogLine(line.str(), endwuid, pID, columnNumPID);
```

Comment:
cleanup: Could now define the foundEndWUID variable here, and remove the assignment below